com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -201,14 +276,45 @@ internal struct AgentParameters
         /// <summary>
         /// Called when the attached <see cref="GameObject"/> becomes enabled and active.
         /// </summary>
+        /// <remarks>
+        /// This function initializes the Agent instance, if it hasn't been initialized yet.
+        /// Always call the base Agent class version of this function if you implement `OnEnable()`
+        /// in your own Agent subclasses.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// protected override void OnEnable()
+        /// {
+        ///     base.OnEnable();
+        ///     // additional OnEnable logic...
+        /// }
+        /// </code>
+        /// </example>
         protected virtual void OnEnable()
         {
             LazyInitialize();
         }
 
         /// <summary>
-        /// <inheritdoc cref="OnBeforeSerialize"/>
+        /// Called by Unity immediately before serializing this object.
         /// </summary>
+        /// <remarks>
+        /// The Agent class uses OnBeforeSerialize() for internal housekeeping. Call the
+        /// base class implementation if you need your own custom serialization logic.
+        ///
+        /// See [OnBeforeSerialize] for more information.
+        /// 
+        /// [OnBeforeSerialize]: https://docs.unity3d.com/ScriptReference/ISerializationCallbackReceiver.OnAfterDeserialize.html
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// public new void OnBeforeSerialize()
+        /// {
+        ///     base.OnBeforeSerialize();
+        ///     // additional serialization logic...
+        /// }
+        /// </code>
+        /// </example>
         public void OnBeforeSerialize()
```

Comment:
Not sure, anyone that didn't go through an intermediate version would get bit by the member move. @surfnerd Any thoughts?

Reply:
1. I don't think we _should_ remove it since we don't know which version people are upgrading from.
2. Is there a dire need to remove this code, or do you just want to?
3.  If we do remove it, we need to call it out in the docs a major change and make people manually change all of their scenes to ensure maxStep is correct.

I feel strongly about not removing it.

Reply:
It is part of the public API. If we can make it private or internal, I would really prefer we do that.

Reply:
I don't think we cane make it private or internal and still have it work. Pretty sure @surfnerd tried that when we were first adding it.

Let's discuss it elsewhere; I don't want to hold up this PR any more than we need to.